packages/debug/src/browser/debug-configuration-manager.ts

```diff
@@ -453,13 +465,72 @@ export class DebugConfigurationManager {
         } catch {
             // Just keep going
         }
+
+        // no configuration yet. Proceed to create the file or use any action registered on the `debug/createConfiguration` menu.
         const debugType = await this.selectDebugType();
         const configurations = debugType ? await this.provideDebugConfigurations(debugType, model.workspaceFolderUri) : [];
+        const commandQuickPicks = this.buildItemsFromMenu();
+
+        if (commandQuickPicks.length > 0) {
+            const quickPickItems: QuickPickItemOrSeparator[] = [];
+            const GENERATE_FILE_ID = 'debug-generate-file';
+            const item: QuickPickItem = {
+                // basic action to provide a launch.json skeleton and user can edit it.
+                label: nls.localize('theia/debug/generateLaunchFile', 'generate launch.json'),
+                id: GENERATE_FILE_ID
+            };
+            quickPickItems.push(item);
+            quickPickItems.push({ type: 'separator' });
+            quickPickItems.push(...commandQuickPicks);
+
+            const selected = await this.quickPickService.show(quickPickItems, { canSelectMany: false });
+            if (selected && selected.id) {
+                if (selected.id === GENERATE_FILE_ID) {
+                    await this.updateTextModel(textModel, configurations);
+                } else {
+                    await this.commandService.executeCommand(selected.id);
+                }
+            }
+        } else {
+            await this.updateTextModel(textModel, configurations);
+        }
+    }
+
+    protected async updateTextModel(textModel: IReference<MonacoEditorModel>, configurations: DebugConfiguration[]): Promise<void> {
         const content = this.getInitialConfigurationContent(configurations);
         textModel.object.textEditorModel.setValue(content); // Will clobber anything the user has entered!
         await textModel.object.save();
     }
 
+    protected buildItemsFromMenu(): QuickPickItemOrSeparator[] {
+        const menu = this.menuRegistry.getMenu(DEBUG_CREATE_CONFIGURATION_MENU);
+        const quickPickItems = [];
+        for (const child of menu.children) {
+            if (child.children) {
```

Comment:
This code duplicates the code from `TabbarToolbarRegistry#visibleItems()`. That code is problematic and I'm not too happy to spread it around more.